src/components/clusterprovider/minikube/minikube.ts

```diff
@@ -198,13 +198,42 @@ async function stopMinikube(context: Context): Promise<void> {
     });
 }
 
+function getStatusFormatString(currentVersion: string): string {
+    if (currentVersion.startsWith("v0.")) {
+        if (parseInt(currentVersion.split(".")[1]) < 31) {
+            // minikube < 0.31.0 uses this format
+            return `"{{.MinikubeStatus}}","{{.ClusterStatus}}","{{.KubeconfigStatus}}"`;
+        } else {
+            // minikube < 1.0.0 uses this format
+            return `"{{.Host}}",{ "kubelet": "{{.Kubelet}}", "apiserver": "{{.ApiServer}}" },"{{.Kubeconfig}}"`;
+        }
+    }
+
+    // minikube >= 1.0.0 uses this format
+    return `"{{.Host}}",{ "kubelet": "{{.Kubelet}}", "apiserver": "{{.APIServer}}" },"{{.Kubeconfig}}"`;
```

Comment:
I think this will result in `JSON.parse` returning an object for the second element of the array?  But the second element gets parsed into `MinikubeInfo.cluster` which is currently declared as `string`.  Can we tweak the `MinikubeInfo` interface to reflect this?  (Though it doesn't look like `cluster` is used anywhere so maybe we could just get rid of it.)

Reply:
the only place where i could find the usage of this code is here: https://github.com/Azure/vscode-kubernetes-tools/blob/7efd603d46e80da9a5c4181143150dc133aedbed/src/extension.ts#L201

it references the first element (`running`), i don't know why other fields are returned though, because they get never used. i just chose to adapt the return value for the newer output accordingly.

i probably could get rid of all the additional elements, just return `.status` (aka a boolean), and probably let status be something like `running: 'Stopped' !== obj[0] && obj[1]`; but i don't know if this is really necessary, and somehow blows up the scope of this pull request.